api/wft-complete-metadata.md

```diff
@@ -0,0 +1,133 @@
+  // data that they'd like to show up the UI, for example.
+  map<string, google.protobuf.Struct> struct_metadata = 1;
+  // For better typed / more efficient binary serialization of types that are expected to be known
+  // by their consumers.
+  map<string, google.protobuf.Any> proto_metadata = 2;
+}
+```
+
+Having both fields seems useful here, since `Any` is useful for the SDK making notes to itself which
+it knows the type of. `Struct` is essentially just a more efficient JSON representation, and can
+be used for arbitrary data that can easily be rendered.
+
+Since `Any` can also be `Struct`, we could have just one field, but this wastes some bytes on the
+`type_url` field if we are throwing a lot of structs in there. Keeping them separate also makes
+it clear what you can decode without any type knowledge, rather than walking the whole map looking
+for struct types in case you want to decode them - that way you only look in the `Any` map when you
+know you're looking for something specific.
+
+### Why not just add more fields to WFT completion response?
+
+* Things that the SDK only needs to write down for itself (ex: the versioning usecase) can now be
+  made without requiring changes to the API repo, updating the server, releasing it, and ensuring
+  that that version is in use. That's real nice.
+* Supports arbitrary user-originated data (TBD if we actually want to allow this, but with this
+  approach it's easy to add if we do).
+
+### Why not use markers?
+
+This is mentioned above, but I want to reiterate it: 
+
+I think it's very important that history events stay focused on recreating workflow state. The
+usecases mentioned here don't impact workflow state, and so they shouldn't get dedicated events
+in history. Anything which *does* directly impact workflow state or the recreation of, should indeed
+be represented as an event.
+
+You might reasonable argue "Hey! The SDK internal versioning usecase you mentioned earlier in some
+sense does or can impact workflow state, why doesn't that get a marker?". I think it's worth
+drawing a distinction between "things that impact workflow state and are a consequence of user
+code", "things that indirectly impact workflow state but are not a consequence of user code" , and
+"things that don't impact state at all". I think only the first of those deserve top-level
+representation as history events. The others can be stored as metadata, or not at all.
+
+## Drawbacks
+* No dedicated fields means no docstrings for those fields, and less type safety. Though this is
+  much less of a concern when the writer and reader are the same person. Type safety can be achieved
+  with the `Any` valued map - and should be used for data that is expected to be interpreted by
+  more than one component.
+* Potential to bloat history.
+
+## Adoption / teaching / etc
+
+Not applicable to this proposal since developers will not *directly* interact with this, but rather
```

Comment:
I think we should definitely show this in the UI and make sure that the additional information is clearly visible.